python/ray/experimental/data/tests/test_dataset.py

```diff
@@ -517,8 +517,7 @@ def test_json_read(ray_start_regular_shared, tmp_path):
     assert df1.equals(ray.get(ds.to_pandas())[0])
     # Test metadata ops.
     assert ds.count() == 3
-    assert "two" in str(ds.schema())
-    assert "two" in str(ds)
+    assert ds.input_files() == [path1]
```

Comment:
Can you add back the previous tests? They are testing schema and str as well.

Reply:
I thought that we're punting on schema fixup for now?

Reply:
No regression here though right?

Reply:
Oh I see, then can you change it to expect "Unknown" schema here?

Reply:
I can change it to expect that a `ValueError` is raised since that's the current behavior, or would we rather return `None` for unset metadata when the dataset metadata accessor methods are called?

Reply:
How about returning None then? To be consistent, we should make the change for `schema()`, `size_bytes()`, and `input_files() [empty list]`.

Reply:
Yeah that makes more sense to me than raising an error.